### DISCUSSION TITLE
ヒューマノイドロボットの要求仕様案

### DISCUSSION THREAD

**Discussion by Yusuke-Yamasaki-555** [open-rdc/Bipedal_Robot_Hardware#1] [Ideas]
各自、自分の考えた仕様を書いていくこと

**Comment by HarukiOgawa1** [comment]
小川
https://docs.google.com/document/d/1cj6gVXa01JaeDsBxIMbtmLHH-2itXTR-Kh301dn5LLU/edit?usp=sharing

**Comment by Yusuke-Yamasaki-555** [comment]
# 要求仕様の策定

## 目標
- スタートから100m進む。

## 要求
- つくばチャレンジのレギュレーションに沿うハードウェアにする。
- 多少の凹凸を含みタイルで構成される平地で、全方向に歩行ができる。
  - 数値化したほうが良い。条件をつくばチャレンジに限定するなら、ちゃんと測ったほうが良い。
- 障害物を認識して、進行を停止、加えて回避することができる。
  - 何を使う？開発コストを考えたほうが良い。
- システム構成をシンプルにする。
- システムを構成する各要素は、開発難易度が比較的低いものを採用する。
- 遮蔽物がない野外の環境で自己位置推定ができる。
- SLAMができる必要はない。
  - GNSSなら、緯度経度の位置をランドマークとして、そこを通るように移動するようにすればよい。地図はいらない。
- 障害物や地面に対して、形状や色の認識を行わず、自己位置推定と2足歩行ができる。

## 構成
- motor
  - steadywin 8108-8 (足りないかも？)
  - 少なくとも、通常で20~30~40Nmあったほうが良い。TRON1、Bercley大のやつを参考にしたい。サイズも同じぐらい。
- computer
  - main: 自己位置推定、GNSS。raspi5?
  - sub: 両脚。組み込み。
- sensor
  - IMU: 安定化。オドメトリ
  - GNSS: 自己位置推定。どれぐらいずれるだろうか？
  - 2D lidar: 障害物検知・回避。斜めにつけて、前方数十cm前の地形を常に見る。んで、ある閾値を超える凹凸を検出したら、それを避ける？
  - エンコーダ
- etc, etc, 中身はMEVIUS準拠で、２足歩行ロボットでいいかも？

**Comment by tomoki057** [comment]
要求仕様
=======================================
目標
１００M~２００M歩行
=======================================
必要な機能

・バランス制御機能
ジャイロセンサー、加速度センサー、IMU

・高トルクモータ
・歩行経路計画および地形認識
LiDAR

・自己位置推定
・衝撃吸収機構
・防水・防塵性能

いらない機能

・転倒復帰
・高速歩行

**Comment by HarukiOgawa1** [comment]
１２月１２日のまとめ

- アクチュエータに関してはコスト的にsteadywin 8108-8を考えるが、膝関節のようなトルクが必要な箇所に関しては8108-8ではなく、steadywinの他のものを使う

- 要求仕様に関しては提案したものをまとめて今回の開発で必要なものを定める

**Comment by yasuohayashibara** [comment]
> アクチュエータに関してはコスト的にsteadywin 8108-8を考えるが、膝関節のようなトルクが必要な箇所に関しては8108-8ではなく、steadywinの他のものを使う

これを議論するためには，膝関節にどの程度トルクが必要なのかを突き止める必要があります．
設計を経験する良い機会だと思いますので，どなたか計算してみると良いと思います．

**Comment by Yusuke-Yamasaki-555** [comment]
## 膝関節に最低限必要なトルクの推定
- 静力学のみで考える。
- Pitch軸のみ考慮し、X-Z軸からなる２次元平面を考える。
- 質量は複数の質点で扱い、腰中心、腰膝間リンク（リンク１）中心、膝足首間リンク（リンク２）中心に置く。
- パラメータは、各質点の質量 [kg] 、各リンク長さ [m] 、各関節角度 [deg] からなる。
- プログラムとモデルは以下を用いる。（手書きのモデルと式が含まれた画像と、それをおこしたpythonプログラムが含まれている）
[humanoid_model_torque_check.zip](https://github.com/user-attachments/files/18147239/humanoid_model_torque_check.zip)

---
### 想定１
- 質量
  - 腰：9
  - リンク１：0.5
  - リンク２：0.5
- 角度（右手系。90 [deg] で床に対して垂直）
  - 腰関節：90
  - 膝関節：65
  - 足関節：115
- リンク長さ
  - 腰重心間リンク：0.1
  - リンク１：0.3
  - リンク２：0.3

結果：各関節トルク [Nm] (右手系なので、負の値は時計回りの回転方向を持つことを示す)
- 腰関節：  -5.40620
- 膝関節：-11.50483
- 足関節：   0.62188
---
### 想定２

**Comment by Yusuke-Yamasaki-555** [comment]
# 要求仕様（決定）

- 100m歩ける。
- 最大時速３km で歩行できる。
- 回路構成は、MEVIUSを参考にする。
- 転倒を防止できるようにする。
- 転倒時の衝撃を吸収できるようにする。
- つくばチャレンジのレギュレーションに沿うハードウェアにする。
- 安全性を担保する。
- 多少の凹凸を含みタイルで構成される平地で、全方向に歩行ができる。
  - 数値化したほうが良い。条件をつくばチャレンジに限定するなら、ちゃんと測ったほうが良い。
  - 障害物を認識して、進行を停止、加えて回避することができる。
    - 何を使う？開発コストを考えたほうが良い。
- システム構成をシンプルにする。
- システムを構成する各要素は、開発難易度が比較的低いものを採用する。
- 遮蔽物がない野外の環境で自己位置推定ができる。
- SLAMができる必要はない。
  - GNSSなら、緯度経度の位置をランドマークとして、そこを通るように移動するようにすればよい。地図はいらない。
- 障害物や地面に対して、形状や色の認識を行わず、自己位置推定と2足歩行ができる。

---

議論の結果、以上の要求仕様を踏まえてロボットの設計を進めていくこととする。

以上でcloseする。設計の議論に移行する。
